### PR TITLE
Start generating own manifest

### PR DESCRIPTION
**Pull request by guifromrio**
First step to allow importing of manifests from other sites / libs on any deco site.

### CODE REVIEW

File: dev.ts [hunk 1]
```diff
@@ -75,13 +78,41 @@ export default async function dev(
...
-    collectFunctions(dir),
+    collectSections(dir).then(mapWith("./sections")),
+    collectFunctions(dir).then(mapWith("./functions")),
   ]);
-  const schemas = await extractAllSchemas(sections, functions);
+
+  let schemas = await extractAllSchemas(sections, functions);
+
+  // "imports" is of format { "nameOfImport" : manifest }
+  for (const [key, importManifest] of Object.entries(imports)) {
```
Comment (igorbrasileiro) [inline dev.ts:96]: have you took a look here at the workbench API? It's a sanity check
https://github.com/deco-cx/live/blob/main/utils/workbench.ts#L21

File: functions/EffectSelectPage.ts [hunk 2]
```diff
@@ -5,6 +5,10 @@ export interface Props {
   pageIds: number[];
 }
 
+export interface Return {
+  selectedPageIds: number[];
+}
+
```
Comment (igorbrasileiro) [inline functions/EffectSelectPage.ts:11]: Does this interface is used?

File: functions/MatchDate.ts [hunk 3]
```diff
@@ -12,11 +12,10 @@ const MatchDate: MatchFunction<Props> = (
   props,
 ) => {
   const now = new Date();
-  const start = new Date(props.start);
-  const end = new Date(props.end);
-  const isMatch = now > start && now < end;
+  const start = props.start ? now > new Date(props.start) : true;
+  const end = props.end ? now < new Date(props.end) : true;
```
Comment (igorbrasileiro) [inline functions/MatchDate.ts:16]: Idk if others think like me and if it's make sense, but i prefer take a look in a interval comparison like this: `lower value < higher value`. Using this for both sentences `lower value < mid value < higher value`

File: sections/Head.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import { asset, Head } from "$fresh/runtime.ts";
```
Comment (igorbrasileiro) [inline sections/Head.tsx:1]: Does this component here is intentional?

**Review by igorbrasileiro — APPROVED** [review]

File: functions/EffectSelectPage.ts (hunk 2 above)
Comment (guifromrio) [reply functions/EffectSelectPage.ts:11]: No but I would like to use it in the future! I think Candeia will solve it with the new engine

File: sections/Head.tsx (hunk 4 above)
Comment (guifromrio) [reply sections/Head.tsx:1]: Yes

**Review by guifromrio — COMMENTED** [review]
LGTM after changes

File: dev.ts [hunk 5]
```diff
@@ -298,3 +334,10 @@ async function extractAllSchemas(
 
   return schemas;
 }
+
+// Generate live own manifest data so that other sites can import native functions and sections.
```
Comment (lucis) [inline dev.ts:338]: Noice!

**Review by lucis — APPROVED** [review]
@guifromrio can you write an article about this before releasing?

File: sections/Head.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  faviconUrl,
+  styleUrls,
+  inlineStyles,
+  scriptUrls,
+  inlineScripts,
+  themeColor,
+}: Props) {
+  return (
+    <Head>
+      <title>{title}</title>
```
Comment (tlgimenes) [inline sections/Head.tsx:30]: O react-helmet tem um "title template" que geramente é bem util. ref: https://github.com/nfl/react-helmet#reference-guide

**Comment by tlgimenes** [comment]
LGTM, however I feel more and more we should either split the runtime from the components or merge everything into the same repo

**Comment by igorbrasileiro** [comment]
> LGTM, however I feel more and more we should either split the runtime from the components or merge everything into the same repo

@guifromrio i don't have enough context, fell free to answer it.

File: sections/Head.tsx (hunk 6 above)
Comment (guifromrio) [reply sections/Head.tsx:30]: Vc acha necessario isso melhorar nesse PR?

**Comment by guifromrio** [comment]
> LGTM, however I feel more and more we should either split the runtime from the components or merge everything into the same repo

I would prefer to merge everything here

**Pull request merged**